libkbfs/data_types.go

```diff
@@ -195,6 +195,9 @@ const (
 	// FilesWithHolesDataVer is the data version for files
 	// with holes.
 	FilesWithHolesDataVer DataVer = 2
+	// BigFilesDataVer is the data version for files that have
+	// multiple levels of indirection.
+	BigFilesDataVer DataVer = 3
```

Comment:
I was thinking about this a bit more, and I think the semantics of `DataVer` are sufficiently subtle enough to warrant an explanatory comment somewhere around here.

Here's my understanding; let me know if I get something wrong:

1) `DataVer` is a per-block attribute, not per-file. This means that, in theory, an indirect block of with `DataVer` `n` may point to blocks with `DataVer`s less than, equal to, or greater than `n`. However, for now, it's guaranteed that an indirect block will never point to blocks with greater versions than itself. (See #3 for details.)

2) `DataVer` is an external attribute of a block, meaning that it's not stored as part of the block, but computed by the creator (or anyone with the latest kbfs client), and stored only in pointers to the block.

2.5) A file (or, in the future a dir) can in theory have any arbitrary tree structure of blocks. However, we only write files such that all paths to leaves have the same depth. (This may already be in a comment somewhere.)

3) Currently, in addition to 2.5, we have the following constraints on tree structures:

a) Direct blocks are always v1.
b) Indirect blocks of depth 2 (meaning one indirect block pointing to all direct blocks) can be v1 (if it has no holes), v2 (if it has holes), or v3 (if it has at least one indirect ptr with `TargetBlockType = Direct` [although if it has one, all of them should be]). However, all its indirect pointers will have `DataVer` 1, by a).
c) Indirect blocks of depth 3 must be v3 and must have at least one indirect pointer with non-zero `TargetBlockType` [although if it holds for one, it should hold for all], although its indirect pointers may have any combination of `DataVer` 1, 2, or 3, by b).
d) Indirect blocks of dept k > 3 must be v3 and must have at least one ... non-zero `TargetBlockType` [... for all], and all of its indirect pointers must have `DataVer` 3, by c).